### PR TITLE
Fix `model_accepts_loss_kwargs` for timm model

### PR DESCRIPTION
**Pull request by qubvel**
# What does this PR do?

`TimmWrapper` model training is broken due to the `num_items_in_batch` param passed to the model forward. `TimmWrapper` model has `**kwargs` in the `transformers` model forward, but it goes straight to the `timm` model and not to the loss. This PR introduces a fix to avoid passing loss_kwargs even if `**kwargs` are in model forward.

Relevant to:
 - https://github.com/huggingface/transformers/pull/35121
 - https://github.com/huggingface/transformers/pull/34915

cc @ArthurZucker

### CODE REVIEW

**Review by ArthurZucker — APPROVED** [review]
Okay, I think we might want to document  `accepts_loss_kwargs` as it's scope right now is super small. 
This can't be set to `True` by default, we can just add a property that checks the forward signature, unless `_accept_loss_kwargs` is set to `false`.

**Comment by qubvel** [comment]
@ArthurZucker Do you mean to move it to the `PretrainedModel` level?

```python

class PreTrainedModel:

    _accept_loss_kwargs = None
    
    @property
    def accepts_loss_kwargs(self):
        if self._accept_loss_kwargs is not None:
            return self._accept_loss_kwargs
        # otherwise check for signature here
        ...
        return accept_loss_kwargs
```

And then in modeling:

```python
class MyModel:
    _accept_loss_kwargs = False
```

**Comment by ArthurZucker** [comment]
No for now let's merge as is! Not sure we'll need the abstraction for more models!
